vortex-array/src/compute/conformance/binary_numeric.rs

```diff
@@ -41,44 +102,333 @@ where
+            PType::I16 => test_binary_numeric_edge_cases_for_type::<i16>(array),
+            PType::I32 => test_binary_numeric_edge_cases_for_type::<i32>(array),
+            PType::I64 => test_binary_numeric_edge_cases_for_type::<i64>(array),
+            PType::U8 => test_binary_numeric_edge_cases_unsigned::<u8>(array),
+            PType::U16 => test_binary_numeric_edge_cases_unsigned::<u16>(array),
+            PType::U32 => test_binary_numeric_edge_cases_unsigned::<u32>(array),
+            PType::U64 => test_binary_numeric_edge_cases_unsigned::<u64>(array),
+            PType::F16 => {
+                eprintln!("Skipping f16 edge case tests (not supported)");
+            }
+            PType::F32 => test_binary_numeric_edge_cases_float::<f32>(array),
+            PType::F64 => test_binary_numeric_edge_cases_float::<f64>(array),
+        },
+        _ => {
+            vortex_panic!(
+                "Binary numeric edge case tests are only supported for primitive numeric types"
+            );
+        }
+    }
+}
+
+fn test_binary_numeric_edge_cases_for_type<T>(array: ArrayRef)
+where
+    T: NativePType + Num + Copy + std::fmt::Debug + num_traits::Bounded + num_traits::Signed,
+    Scalar: From<T>,
+{
+    // Test with zero
+    test_binary_numeric_with_scalar(array.clone(), T::zero());
+
+    // Test with -1
+    test_binary_numeric_with_scalar(array.clone(), -T::one());
+
+    // Test with max value
+    test_binary_numeric_with_scalar(array.clone(), T::max_value());
+
+    // Test with min value
+    test_binary_numeric_with_scalar(array, T::min_value());
+}
+
+fn test_binary_numeric_edge_cases_unsigned<T>(array: ArrayRef)
+where
+    T: NativePType + Num + Copy + std::fmt::Debug + num_traits::Bounded,
+    Scalar: From<T>,
+{
+    // Test with zero
+    test_binary_numeric_with_scalar(array.clone(), T::zero());
+
+    // Test with max value
+    test_binary_numeric_with_scalar(array.clone(), T::max_value());
+
+    // Test with min value (0 for unsigned)
+    test_binary_numeric_with_scalar(array, T::min_value());
+}
+
+fn test_binary_numeric_edge_cases_float<T>(array: ArrayRef)
+where
+    T: NativePType + Num + Copy + std::fmt::Debug + num_traits::Float,
+    Scalar: From<T>,
+{
+    // Test with zero
+    test_binary_numeric_with_scalar(array.clone(), T::zero());
+
+    // Test with -1
+    test_binary_numeric_with_scalar(array.clone(), -T::one());
+
+    // Test with max value
+    test_binary_numeric_with_scalar(array.clone(), T::max_value());
+
+    // Test with min value
+    test_binary_numeric_with_scalar(array.clone(), T::min_value());
+
+    // Test with small positive value
+    test_binary_numeric_with_scalar(array.clone(), T::epsilon());
+
+    // Test with special float values (NaN, Infinity)
+    test_binary_numeric_with_scalar(array.clone(), T::nan());
+    test_binary_numeric_with_scalar(array.clone(), T::infinity());
+    test_binary_numeric_with_scalar(array, T::neg_infinity());
```

Comment:
If we're enumerating all the funny guys, it seems we should include `T::min_positive_value()` which is a subnormal floating-point number